sismic/io/datadict.py

```diff
@@ -130,7 +130,15 @@ def _import_state_from_dict(state_d: Mapping[str, Any]) -> StateMixin:
         if substates and parallel_substates:
             raise StatechartError('{} cannot declare both a "states" and a "parallel states" property'.format(name))
         elif substates:
-            state = CompoundState(name, initial=state_d.get('initial', None), on_entry=on_entry, on_exit=on_exit)
+            initial = state_d.get('initial', None)
+            if isinstance(initial, dict):
+                try:
+                    initial = _import_transition_from_dict(name, initial)
```

Comment:
We should be careful here, since `name` is used to populate the `source` attribute of a `Transition`, it is likely that any initial transition will confuse `Interpreter._select_transitions` (i.e. they will be selected because the source state is active). I don't know how to address this "major" issue, perhaps by setting the `source` of an initial transition to `None` (it makes sense, since this transition comes "out of the blue" to some extent).

Reply:
Yes setting the source to None or (null in the UML) is the appropriate way. Sorry I haven't got around to changing anything it won't be until later after the kids are in bed that I fix these issues.